frontend/src/api/metadata.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * API functions for calculating metadata histograms.
+ * We separate them into getting buckets, counts, and metrics so we
+ * can run then asynchronously and provide interactive updates while waiting
+ * for more expensive computations like calculating metrics.
+ */
+import { ZenoColumnType } from "../globals";
+import { InternMap } from "internmap";
+import { metricRange } from "../stores";
+import { columnHash, getMetricRange } from "../util/util";
+import { get } from "svelte/store";
+
+/**
+ * Fetch metadata columns buckets for histograms.
+ *
+ * @param completeColumns All complete columns.
+ * @param model Current model.
+ *
+ * @returns Histogram buckets for each column.
+ */
+export async function getHistograms(
+	completeColumns: ZenoColumn[],
+	model: string
+): Promise<InternMap<ZenoColumn, HistogramEntry[]>> {
+	const requestedHistograms = completeColumns.filter(
+		(c) =>
+			(c.model === "" || c.model === model) &&
+			c.columnType !== ZenoColumnType.OUTPUT
+	);
+	const res = await fetch("/api/histograms", {
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify(requestedHistograms),
+	}).then((res) => res.json());
+	const histograms = new InternMap<string, HistogramEntry[]>(
+		requestedHistograms.map((col, i) => [col, res[i]]),
+		columnHash
+	);
+	return histograms;
+}
+
+let histogramCountAborter = null;
```

Comment:
This is probably a me problem. I am thoroughly confused what this aborter system is doing.